companion/gateway/security/defensive_response.py

```
@@ -0,0 +1,345 @@
+        extraversion = big_five.get("extraversion", 0.5)
+        agreeableness = big_five.get("agreeableness", 0.5)
+        neuroticism = big_five.get("neuroticism", 0.5)
+        
+        # Get PAD state for emotional consistency
+        pleasure = current_pad.get("pleasure", 0.0)
+        arousal = current_pad.get("arousal", 0.0)
+        dominance = current_pad.get("dominance", 0.0)
+        
+        # Sanitize quirk names to prevent prompt injection
+        safe_quirks = []
+        for q in active_quirks:
```

Ruff (0.13.2)

Undefined name `active_quirks`

(F821)